orderbook/src/fee.rs

```diff
@@ -248,18 +258,24 @@ impl MinFeeCalculating for MinFeeCalculator {
     }
 
     // Returns true if the fee satisfies a previous not yet expired estimate, or the fee is high enough given the current estimate.
-    async fn is_valid_fee(&self, sell_token: H160, fee: U256) -> bool {
+    async fn is_valid_fee(&self, sell_token: H160, fee: U256, app_data: [u8; 32]) -> bool {
+        let app_based_fee_factor = self
+            .partner_fee_factors
+            .get(&H256::from(app_data))
+            .unwrap_or(&1.0);
+        let scaled_fee = U256::from_f64_lossy(fee.to_f64_lossy() / app_based_fee_factor);
```

Comment:
One thing I'm not a huge fan of is that we apply fee factors in two different places. Once here, and once in `compute_min_fee`. Maybe we can refactor `compute_min_fee` to also take the `app-data` and do all the fee subsidy computation in one place?

Reply:
This scaled fee is only computed once... I can make your suggestion happen anyway, but I don't see how it's being applied twice at the moment.

Reply:
Sorry, my comment wasn't super clear. I meant we apply fee factors in two different places in code (the global one, and the partner one). They are both correctly only applied once, the comment was more about how it would be nice if the fee factor computations only appear in a single place in the code, to make it easier to find and maintain (so more a code maintenance comment, not a correctness one).

However, looking at the code, this might be a problem because the app ID isn't known at the time the "saved" min fee is created.